File: web/src/types/dispositionSettings.ts
export type KuehlraumWandTab = 'kuehlraum' | 'extern';

export type PersonnelRole = 'arrangeur' | 'traeger' | 'fahrer';

/** Person im Disposition-Personalpool (Arrangeur / Träger / Fahrer). */
export interface DispositionPerson {
  id: string;
  name: string;
  roles: PersonnelRole[];
  active?: boolean;
  /** Externe Person (z. B. Träger von außerhalb) — eigener Tab in der Einbuchung. */
  extern?: boolean;
}

export interface EigenerKuehlraumConfig {
  id: string;
  label: string;
  /** Anzeige in Alamida, z. B. „Kühlr. Grafenbach“ */
  alamidaName?: string;
  /** Erkennung in Ortsnamen (Kleinbuchstaben, enthält) */
  matchKeywords: string[];
  /** Externe Abholorte (UK, Senecura …) → Zuordnung zu diesem Kühlraum */
  externKeywords: string[];
  /** Wandmonitor: Tab „Kühlraum“ (Platzraster) oder „Extern“ (Kartenliste) */
  wandTab?: KuehlraumWandTab;
  plaetze: number;
  /** Planung: Tage seit Freigabe bei Belegung anzeigen (Freigabetag zählt mit). */
  zeigeTageSeitFreigabe?: boolean;
  /** Planung: Belegung auch in der linken Spalte („Aktuelle Orte“) anzeigen. */
  zeigeInLinkerPlanungsspalte?: boolean;
}

/** Anzeigedauer je Wandmonitor-Tab vor dem automatischen Wechsel (Sekunden). */
export interface WallTabWechselSekunden {
  kuehlraum: number;
  extern: number;
  kalender: number;
  abholungen: number;
  offen: number;
}

/** Aktiviert Tabs für den automatischen Rotationslauf am Wandmonitor. */
export interface WallTabRotationEnabled {
  kuehlraum: boolean;
  extern: boolean;
  kalender: boolean;
  abholungen: boolean;
  offen: boolean;
}

export interface DispositionSettings {
  kremationPrefixe: string[];
  kremationKeywords: string[];
  krankenhausPrefixe: string[];
  krankenhausKeywords: string[];
  /** Extern-Wand: Pflegeheim / Senecura … */
  pflegeheimPrefixe: string[];
  pflegeheimKeywords: string[];
  /** Extern-Wand: Bestattung / Bestatter … */
  bestattungPrefixe: string[];
  bestattungKeywords: string[];
  eigeneKuehlraeume: EigenerKuehlraumConfig[];
  /** Pool für Kalender-Personal-Einbuchung (Arrangeur / Träger / Fahrer). */
  personnelPool?: DispositionPerson[];
  /** Wandmonitor: Sekunden pro Tab bis zum nächsten Übergang */
  wallTabWechselSekunden?: WallTabWechselSekunden;
  /** Wandmonitor: welche Tabs in der Rotation angezeigt werden */
  wallTabRotationEnabled?: WallTabRotationEnabled;
  updatedAt?: { seconds: number };
  /** Änderungszähler für Agent-Reload */
  settingsVersion?: number;
}

File: agent/Alamida.Monitoring.Core/Models/DispositionSettings.cs
namespace Alamida.Monitoring.Core.Models;

public sealed class DispositionSettings
{
    public List<string> KremationPrefixe { get; init; } = [];
    public List<string> KremationKeywords { get; init; } = [];
    public List<string> KrankenhausPrefixe { get; init; } = [];
    public List<string> KrankenhausKeywords { get; init; } = [];
    public List<string> PflegeheimPrefixe { get; init; } = [];
    public List<string> PflegeheimKeywords { get; init; } = [];
    public List<string> BestattungPrefixe { get; init; } = [];
    public List<string> BestattungKeywords { get; init; } = [];
    public List<EigenerKuehlraumConfig> EigeneKuehlraeume { get; init; } = [];

    public static DispositionSettings Default => new()
    {
        KremationPrefixe = [],
        KremationKeywords =
        [
            "krematorium", "innermanzing", "feba", "kremation", "einäscherung",
            "einaescherung", "feuerbestattung", "einäscherungsanlage",
        ],
        KrankenhausPrefixe = ["UK ", "UK-", "KH ", "KH-", "KH."],
        KrankenhausKeywords = ["krankenhaus", "spital", "klinik", "landesklinik"],
        PflegeheimPrefixe = ["Senecura"],
        PflegeheimKeywords = ["senecura", "pflegeheim", "altenheim", "hospiz", "pflege"],
        BestattungPrefixe = [],
        BestattungKeywords = ["bestattung", "bestatter", "trauerhalle", "bestattungsinstitut"],
        EigeneKuehlraeume =
        [
            new EigenerKuehlraumConfig
            {
                Id = "grafenbach",
                Label = "Firmenkühlraum Grafenbach",
                AlamidaName = "Kühlr. Grafenbach",
                MatchKeywords =
                [
                    "grafenbach", "kühlr. grafenbach", "kuehlr. grafenbach",
                    "kühl. grafenbach", "kuehl. grafenbach",
                ],
                Plaetze = 9,
            },
        ],
    };
}

public sealed class EigenerKuehlraumConfig
{
    public string Id { get; init; } = "";
    public string Label { get; init; } = "";
    public string? AlamidaName { get; init; }
    public List<string> MatchKeywords { get; init; } = [];
    public List<string> ExternKeywords { get; init; } = [];
    public int Plaetze { get; init; } = 9;
}
